fix(messages): add append parameter to history entries

luukvbaal opened the pull request:
Problem:  The "append" parameter added in abb40ece is missing from
          history entries, resulting in different message formatting
          for "g<".
Solution: Add "append" field to message history entries.

Co-authored-by: phanium <91544758+phanen@users.noreply.github.com>

justinmk commented on runtime/doc/lua.txt:
```diff
@@ -2697,14 +2697,14 @@ There are four separate window types used by this interface:
+• "pager": The pager window; used for |:messages| and certain messages that
+  should be shown in full.
+
+These four windows are assigned the "Cmdline", "Messages", "Dialog" and
+"Pager" 'filetype' respectively. Use a |FileType| autocommand to configure any
```
filetype names are usually lowercase. though the buffer names would pascalcase (`[Cmdline]`, etc)

luukvbaal replied:
Sure, changing that in #34378 then. Will rebase when that is merged.

justinmk replied:
just for reference: it looks like there is a [`messages` filetype](https://github.com/neovim/neovim/blob/d86d4bacc14bc6ab51b3173f3644d703f53ffbde/runtime/lua/vim/filetype.lua#L2080-L2081) already :(  so we may need to think of a different filetype name for that one. can name it `messages2` and revisit later.

luukvbaal committed:
fix(messages): add append parameter to history entries

Problem:  The "append" parameter added in abb40ece is missing from
          history entries, resulting in different message formatting
          for "g<".
Solution: Add "append" field to message history entries.

Co-authored-by: phanium <91544758+phanen@users.noreply.github.com>

justinmk commented on src/nvim/message_defs.h:
```diff
@@ -18,4 +18,5 @@ typedef struct msg_hist {
   HlMessage msg;          ///< Highlighted message.
   const char *kind;       ///< Message kind (for msg_ext)
   bool temp;              ///< Temporary message since last command ("g<")
+  bool append;            ///< msg_ext_append
```
a slightly more semantic comment?
